sympy/stats/rv.py

```diff
@@ -1076,57 +1079,96 @@ def sample(expr, condition=None, size=(), library='scipy',
         No modifications to environment's global seed settings
         are done by this argument.
 
+    .. versionchanged:: 1.7.
+        sample returns an iterator containing the samples instead of value
+
+    .. versionchanged:: 1.9.
+        sample returns values or array of values instead of an iterator and numsamples is deprecated.
+
     Examples
     ========
 
     >>> from sympy.stats import Die, sample, Normal, Geometric
     >>> X, Y, Z = Die('X', 6), Die('Y', 6), Die('Z', 6) # Finite Random Variable
+    >>> die_roll = sample(X + Y + Z)
+    >>> die_roll # doctest: +SKIP
+    3
 
-    >>> die_roll = sample(X + Y + Z) # doctest: +SKIP
-    >>> next(die_roll) # doctest: +SKIP
-    6
     >>> N = Normal('N', 3, 4) # Continuous Random Variable
-    >>> samp = next(sample(N)) # doctest: +SKIP
-    >>> samp in N.pspace.domain.set # doctest: +SKIP
+    >>> samp = sample(N)
+    >>> samp in N.pspace.domain.set
     True
-    >>> samp = next(sample(N, N>0)) # doctest: +SKIP
-    >>> samp > 0 # doctest: +SKIP
+    >>> samp = sample(N, N>0)
+    >>> samp > 0
     True
-    >>> samp_list = next(sample(N, size=4)) # doctest: +SKIP
-    >>> [sam in N.pspace.domain.set for sam in samp_list] # doctest: +SKIP
+    >>> samp_list = sample(N, size=4)
+    >>> [sam in N.pspace.domain.set for sam in samp_list]
     [True, True, True, True]
+
+    >>> sample(N, numsamples=3) # doctest: +SKIP
+    [6.07745952932677, 8.024735984727329, 3.7581166319235315]
```

Comment:
If `numsamples` is deprecated then examples like this should go in a section that clearly explains that it is deprecated.